chromium_src/chrome/browser/profiles/profile_avatar_icon_util.cc

```diff
@@ -0,0 +1,208 @@
+      BraveRewriteIconIndexForPlaceholder(index));
+}
+
+std::string GetDefaultAvatarIconUrl(size_t index) {
+  return GetDefaultAvatarIconUrl_ChromiumImpl(
+      BraveRewriteIconIndexForPlaceholder(index));
+}
+int GetDefaultAvatarLabelResourceIDAtIndex(size_t index) {
+  return GetDefaultAvatarLabelResourceIDAtIndex_ChromiumImpl(
+      BraveRewriteIconIndexForPlaceholder(index));
+}
+
+size_t GetBraveAvatarIconStartIndex() {
+  return kDefaultAvatarIconsCount - kBraveDefaultAvatarIconsCount + 1;
+}
+
+const IconResourceInfo* GetBraveDefaultAvatarIconResourceInfo(
+      size_t index) {
+  CHECK_LT(index, kBraveDefaultAvatarIconsCount);
+  // NOTE(petemill): Whilst new profiles get assigned random icons,
+  //  the initial profile for Brave always gets the placeholder item
+  //  (index 56).
+  //  We keep the entry for this first item here, so it is still selectable
+  //  in the Settings and Profile Manager WebUI.
+  // Keep the chromium naming style for compatibility with format compare
+  // methods, such as profile::IsDefaultAvatarIconUrl.
+  // The ID suffixes should be the only thing that needs to change in this file
+  // if the chromium list grows in size.
+  // Generated via zsh:
+  // NOLINT > declare -i a=56
+  // NOLINT > for file in $(cat < avatarlist); do echo "{IDR_PROFILE_AVATAR_$a, \"$file\",
+  //     IDS_BRAVE_AVATAR_LABEL_$a},"; a=a+1 done | pbcopy
+  static const IconResourceInfo resource_info[kBraveDefaultAvatarIconsCount] = {
+    {IDR_PROFILE_AVATAR_56, "avatar_braveorange.png",
+            IDS_BRAVE_AVATAR_LABEL_56},
+    {IDR_PROFILE_AVATAR_57, "avatar_orangepeel.png",
+            IDS_BRAVE_AVATAR_LABEL_57},
+    {IDR_PROFILE_AVATAR_58, "avatar_nana.png",
+            IDS_BRAVE_AVATAR_LABEL_58},
+    {IDR_PROFILE_AVATAR_59, "avatar_envy.png",
+            IDS_BRAVE_AVATAR_LABEL_59},
+    {IDR_PROFILE_AVATAR_60, "avatar_clearskies.png",
+            IDS_BRAVE_AVATAR_LABEL_60},
+    {IDR_PROFILE_AVATAR_61, "avatar_bondyblue.png",
+            IDS_BRAVE_AVATAR_LABEL_61},
+    {IDR_PROFILE_AVATAR_62, "avatar_blurple.png",
+            IDS_BRAVE_AVATAR_LABEL_62},
+    {IDR_PROFILE_AVATAR_63, "avatar_darknight.png",
+            IDS_BRAVE_AVATAR_LABEL_63},
+    {IDR_PROFILE_AVATAR_64, "avatar_orangepeel_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_64},
+    {IDR_PROFILE_AVATAR_65, "avatar_nana_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_65},
+    {IDR_PROFILE_AVATAR_66, "avatar_envy_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_66},
+    {IDR_PROFILE_AVATAR_67, "avatar_clearskies_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_67},
+    {IDR_PROFILE_AVATAR_68, "avatar_bondyblue_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_68},
+    {IDR_PROFILE_AVATAR_69, "avatar_blurple_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_69},
+    {IDR_PROFILE_AVATAR_70, "avatar_darknight_shape.png",
+            IDS_BRAVE_AVATAR_LABEL_70},
+  };
+  return &resource_info[index];
```

Comment:
can we put a `static assert` to make sure the last item we get is actually equal to `resource_info [kBraveDefaultAvatarIconsCount - 1]`. It should look like this
```
 static constexpr IconResourceInfo resource_info[kBraveDefaultAvatarIconsCount] = {
 ....
 {IDR_PROFILE_AVATAR_70, "avatar_darknight_shape.png", IDS_BRAVE_AVATAR_LABEL_70},
};

static_assert((resource_info[kBraveDefaultAvatarIconsCount - 1].resource_id == IDR_PROFILE_AVATAR_70),
    "Entries less than kBraveDefaultAvatarIconsCount");
```
If we put more than `kBraveDefaultAvatarIconsCount` items, we will get compile error but if we somehow accidentally remove some entries and forgot to decrease `kBraveDefaultAvatarIconsCount`, this won't get caught by compiler. And that give us the opportunity to access uninitialized item which will lead to crash like accessing filename.

Reply:
Great suggestion - that's done now